app/src/main/java/org/greenstand/android/TreeTracker/managers/Locations.kt

```diff
@@ -0,0 +1,329 @@
+package org.greenstand.android.TreeTracker.managers
+
+import android.content.Context
+import android.content.pm.PackageManager
+import android.location.Location
+import android.location.LocationManager
+import android.os.Bundle
+import androidx.core.content.ContextCompat
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+import com.google.gson.GsonBuilder
+import java.util.Deque
+import java.util.LinkedList
+import java.util.UUID
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.launch
+import org.greenstand.android.TreeTracker.database.TreeTrackerDAO
+import org.greenstand.android.TreeTracker.database.entity.LocationDataEntity
+import org.greenstand.android.TreeTracker.utilities.LocationDataConfig
+import org.greenstand.android.TreeTracker.utilities.LocationDataConfig.CONVERGENCE_DATA_SIZE
+import org.greenstand.android.TreeTracker.utilities.LocationDataConfig.LAT_STD_DEV
+import org.greenstand.android.TreeTracker.utilities.LocationDataConfig.LONG_STD_DEV
+import org.greenstand.android.TreeTracker.utilities.ValueHelper
+import timber.log.Timber
+
+class LocationUpdateManager(
+    private val locationManager: LocationManager,
+    private val context: Context
+) {
+
+    private val locationUpdates = MutableLiveData<Location?>()
```

Comment:
This is the code that exists prior to my change. I suspect this is nullable because when the app starts it needs an initial value for the location which in this case is null.